e2e/rescheduling/rescheduling.go

```diff
@@ -374,20 +374,27 @@ func (tc *RescheduleE2ETest) TestRescheduleMaxParallelAutoRevert(f *framework.F)
 		"should have one successful, one failed, and 3 reverted allocs",
 	)
 
+	// at this point the allocs have been checked but we need to wait for the
+	// deployment to be marked complete before we can assert that it's successful
+	// and verify the count of deployments
+	f.NoError(
+		e2e.WaitForLastDeploymentStatus(jobID, ns, "successful", nil),
+		"most recent deployment should be successful")
+
 	out, err := e2e.Command("nomad", "deployment", "status")
 	f.NoError(err, "could not get deployment status")
 
 	results, err := e2e.ParseColumns(out)
 	f.NoError(err, "could not parse deployment status")
-	statuses := []string{}
+	statuses := map[string]int{}
 	for _, row := range results {
 		if row["Job ID"] == jobID {
-			statuses = append(statuses, row["Status"])
+			statuses[row["Status"]]++
 		}
 	}
-	f.True(reflect.DeepEqual([]string{"running", "failed", "successful"}, statuses),
-		fmt.Sprintf("deployment status was: %#v", statuses),
-	)
+
+	f.Equal(1, statuses["failed"], "expected only 1 failed deployment")
+	f.Equal(2, statuses["successful"], "expected 2 successful deployments")
```

Comment:
it might be helpful to dump the content of `statuses` in the error messages here, in case something in this test breaks

Reply:
Good call, will do.

Reply:
Done in d39a4eb